### ISSUE TITLE
ivy does not give error when component registered with multiple modules

### ISSUE THREAD

**Issue by santoshyadav198613** [angular/angular#32477]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/core

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....
Not sure

### Description

 A clear and concise description of the problem...

1. When Ivy is set to true, the ng serve command is not giving any error.
2. In Ivy mode, After saving the files the component gives the error 
"Template error: Can't bind to 'ngForOf' since it isn't a known property of 'li'" and this error only comes with the component which is part of the lazy-loaded module and root module.

In repository given for reproduction navigate to Exams and then save the HTML or ts files and notice the error in console.

## 🔬 Minimal Reproduction

 https://stackblitz.com/...

https://github.com/santoshyadav198613/ivy-error

## 🔥 Exception or Error
<pre><code>

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 8.3.2
Node: 10.14.2
OS: win32 x64
Angular: 8.2.4
... common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.2
@angular-devkit/build-angular     0.803.2
@angular-devkit/build-optimizer   0.803.2
@angular-devkit/build-webpack     0.803.2
@angular-devkit/core              8.3.2
@angular-devkit/schematics        8.3.2
@angular/cdk                      8.1.4
@angular/cli                      8.3.2
@ngtools/webpack                  8.3.2
@schematics/angular               8.3.2
@schematics/update                0.803.2
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2

</code></pre>

**Anything else relevant?**

**Comment by Y-Moshe** [comment]
works fine on my machine, try to upgrade node.js to version 12++ if possible.

```
Angular CLI: 8.3.2
Node: 12.3.1
OS: win32 x64
Angular: 8.2.4
... common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.2
@angular-devkit/build-angular     0.803.2
@angular-devkit/build-optimizer   0.803.2
@angular-devkit/build-webpack     0.803.2
@angular-devkit/core              8.3.2
@angular-devkit/schematics        8.3.2
@angular/cdk                      8.1.4
@angular/cli                      8.3.2
@ngtools/webpack                  8.3.2
@schematics/angular               8.3.2
@schematics/update                0.803.2
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2
```

**Comment by santoshyadav198613** [comment]
> works fine on my machine, try to upgrade node.js to version 12++ if possible.
> 
> ```
> Angular CLI: 8.3.2
> Node: 12.3.1
> OS: win32 x64
> Angular: 8.2.4
> ... common, compiler, compiler-cli, core, forms
> ... language-service, platform-browser, platform-browser-dynamic
> ... router
> 
> Package                           Version
> -----------------------------------------------------------
> @angular-devkit/architect         0.803.2
> @angular-devkit/build-angular     0.803.2
> @angular-devkit/build-optimizer   0.803.2
> @angular-devkit/build-webpack     0.803.2
> @angular-devkit/core              8.3.2
> @angular-devkit/schematics        8.3.2
> @angular/cdk                      8.1.4
> @angular/cli                      8.3.2
> @ngtools/webpack                  8.3.2
> @schematics/angular               8.3.2
> @schematics/update                0.803.2
> rxjs                              6.4.0
> typescript                        3.5.3
> webpack                           4.39.2
> ```

There is no compilation error, it only fails for the scenario mentioned above.

**Comment by alxhub** [comment]
Correct - this error has not been implemented yet in Ivy.

**Comment by alxhub** [comment]
And it was implemented in #34404!
